src/precice/bindings/python_future/error_handling.py

```diff
@@ -0,0 +1,16 @@
+"""
+Functions to check the type of input argument against what is expected
+and raise an Exception if mismatch found
+"""
+
+
+def check1D(func_name, usr_input, true_val):
+    if usr_input != true_val:
+        raise Exception("Function {} received position of type {} but expects a 1D numpy \
+                         array with {} values as position input".format(func_name, usr_input, true_val))
```

Comment:
I like the new functions for checks. However, I think currently this function is trying to address two issues and this makes it a bit confusing.

What do we want to check?

1. Is the input a numpy array or at least something that we can transform into a numpy array? I.e. can we even access `input.shape`?
2. Does the dimensions match? Does the number of vertices match? Be careful, these are two different things and we should also give the correct error depending on the context.

Currently `check1D` and `check2D` are always used for checking the second requirement. The first requirement is not checked, even though, the error message of `check1D` suggests that it is checking for the type (I assume this is a copy-paste-error).

Reply:
Additionally, if you split these two concerns, you can raise a `TypeError`, if the type does not match and a different kind of error if the dimension or size is wrong. `ValueError` might be appropriate, but I am not 100% whether this is matching. At least Scipy raises a `ValueError` for dimension mismatch.

Reply:
I agree that we need to separately check if the input array is a numpy array and hence can we even access functionality like `shape`. But then one question is raised, if the user has not provided a numpy array do we convert it and notify the user of the conversion or do we just throw an error and tell the user to convert it ?

Reply:
I would be strict here: If we do not receive something that we can work with, throw an error.

* We really need `shape`, in the case of 2D data (vector data). Everything else than a `np.array` is invalid input -> throw an error.
* We can also use `len()` instead, for 1D data (scalar data, lists of IDs...). Here, any object that supports `len()` is fine -> accept it, if possible.